test/users/npadmana/fftw/testFFTW.chpl

```diff
@@ -1,29 +1,79 @@
+the code.
+
+To compile this code :
+   chpl testFFTW.chpl
+If FFTW is not on your include/library paths, you can do 
+   chpl testFFTW.chpl -I$FFTW_DIR/include -L$FFTW_DIR/lib
+where $FFTW_DIR points to your FFTW installation.
+
+Note that you do not need to explicitly include the header or library files; the module uses an experimental
+feature of the Chapel compiler to automatically include these.
+
+The FFTW module uses the FFTW3 API, and currently just implements the basic interface.
+We will assume that the reader is familiar with using FFTW; more 
+details are at http://www.fftw.org. 
 
+The code computes a series of 1,2 and 3D transforms, exercising the complex <-> complex, and real<->complex
+transforms (both in and out of place). The output of the code should be a series of small numbers ( <= 10^-13);
+see testFFTW.good for example values (it is possible that your values might differ). 
 
+The data for these tests is in arr{1,2,3}d.dat. The format of these files is documented below.
+*/
+use FFTW;
+
+/* Utility function to print the maximum absolute deviation between values computed by this code, 
+   and "truth". The true values are computed using Mathematica v10 */
 proc printcmp(x, y) {
   var err = max reduce abs(x-y);
-  //writeln("----");
-  //writeln(x);
-  //writeln("**");
-  //writeln(y);
-  //writeln("----");
   writeln(err);
 }
 
+/* This is main test code, parametrized on the number of dimensions ndim. 
+  fn is the file that contains the test data.
+*/
 proc runtest(param ndim : int, fn : string) {
-  /* Read in the data and set up the domains appropriately */
   var dims : ndim*int(32); 
+
+  /* We define a number of different domains below, corresponding to complex input/output arrays,
+     real (input) -> complex (output) out-of-place arrays, and real <-> complex in-place arrays.
+
+     The domains are :
+     * D : for complex <-> complex transforms. This domain is also used for the real array in 
+           a real->complex out-of-place transform.
+     * cD : for the complex array in a real <-> complex out-of-place transform.
+     * rD : for the real array in a real <-> complex in-place transform. This includes the padding needed 
+            by the in-place transform. D is a sub-domain of this, and can be used to extract the real array, 
+            without padding.
+     * reD, imD : Utility domains that access the real/complex parts of the complex array in a real<->complex in-place transform.
+  */
   var D : domain(ndim);
-  // Define ranges here
   var rD,cD,reD,imD : domain(ndim,int,true); 
+  
+
+  /* Read in the arrays from the file below. A,B are the arrays that will be used in the FFTW calls, 
+     goodA and goodB store the true values. 
+
+     Note that goodA is all real, allowing us to reuse the same array for the real<->complex tests.
+
+     The file format is as follows (all little-endian):
```

Comment:
It's a trivial thing but I am mildly surprised that the file format doesn't include the rank; as a minimum it could be used for sanity checking and it opens the door to more flexible dispatching e.g. read the rank from the file and then call the appropriate variation of the code.